bot/exts/fun/user_events.py

```diff
@@ -0,0 +1,603 @@
+        # Remove scheduler related to the scheduled event
+        self.scheduler.cancel(scheduled_event["user_event"]["organizer"])
+
+        # DELETE scheduled event on site
+        await self.bot.api_client.delete(f"bot/scheduled-events/{scheduled_event['id']}")
+
+        # Update user event status
+        status = self.not_scheduled()
+        await self.update_user_event_message(status, scheduled_event["user_event"])
+
+    @group(name="userevent", invoke_without_command=True)
+    async def user_event(self, ctx: Context) -> None:
+        """Commands to perform CRUD operations on user events and scheduled events."""
+        await ctx.send_help(ctx.command)
+
+    @user_event.command(name="create")
+    async def create_user_event(self, ctx: Context, event_name: str, *, event_description: str) -> None:
+        """Create a new user event."""
+        organizer = ctx.author.id
+
+        # Ask user to confirm before event creation
+        confirmation_message, embed = await self.send_confirmation_message(event_name, event_description, ctx.author)
+
+        def check(reaction: Reaction, user: Member) -> bool:
+            """Check for correct reaction and user."""
+            return (
+                user == ctx.author and str(reaction.emoji) in EMOJIS.values()
+                and reaction.message.id == confirmation_message.id
+            )
+
+        # Check if user is OK for event creation
+        try:
+            choice, _ = await self.bot.wait_for('reaction_add', timeout=60.0, check=check)
+        except asyncio.TimeoutError:
+            await self.user_event_coord_channel.send("User Event not created.")
+            return
+
+        # User reacting to `cross` indicates cancellation of event creation process.
+        if str(choice) == EMOJIS["cross"]:
+            await ctx.send("User Event not created.")
+            return
+
+        # POST new user event
+        post_data = {
+            "name": event_name,
+            "organizer": organizer,
+            "description": event_description,
+            "message_id": 0  # patch message_id after sending message in user_events channel
+        }
+        await self.bot.api_client.post("bot/user-events", data=post_data)
+
+        # List new event in user-events-list channel
+        message = await self.list_new_event(embed)
+
+        # PATCH message id of newly created user event
+        patch_data = {
+            "message_id": message.id
+        }
+        await self.bot.api_client.patch(f"bot/user-events/{event_name}", data=patch_data)
+
+        await ctx.send(f"User Event **{event_name}** created.")
+
+    @user_event.command(name="change_desc", aliases=["cd", "desc"])
+    async def change_description(self, ctx: Context, event_name: str, *, event_description: str) -> None:
+        """Change user event description."""
+        # Check if user is event organizer
+        if not self.check_if_user_is_organizer(ctx.author.id, event_name):
```

Comment:
You are not awaiting it and also I think more clean way to do this is do this in decorator.